python/res/enkf/jobs/scaling/enkf_scaling_scaled_matrix.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+
+from copy import deepcopy
+from sklearn.decomposition import PCA
+
+
+class DataMatrix(object):
+    def __init__(self, input_data):
+        """
+        Takes input data in the form of a Pandas dataframe with observations,
+        standard deviation and simulated values. Assumes observations are
+        prepended with _OBS and standard deviation with _STD.
+        """
+        self.data = input_data
+
+    def get_data_matrix(self, observation_keys):
+        """
+        Extracts data from a dataframe given a set of keys and returns a matrix
+        """
+        return self.data.loc[list(observation_keys)].as_matrix()
+
+    def std_normalization(self, observation_keys, inplace=False):
+        """
+        Duplicates the behavior of obs_data_scale, and scales the simulated data
+        by 1 / (observation standard deviation), per observation key, i.e. each
+        simulation data point is scaled by its corresponding std deviation
+        from observations.
+        """
+        output_data = deepcopy(self.data)
+        for key in observation_keys:
+            data_matrix = self.data.loc[key]
+            std_vector = self.data.loc["STD_" + key]
+            output_data.loc[key] = (data_matrix * (1. / std_vector))
+
+        if inplace:
+            self.data = output_data
+        return output_data
+
+
+    def get_scaling_factor(self, events):
+        """
+        Performs PCA, calculates the number of primary components based on
+        a threshold and returns a scaling factor based on the number of
+        primary components and the number of observations.
+        """
+        data_matrix = self.get_data_matrix(events.keys)
+        primary_components = self._pca_analysis(data_matrix)
+        nr_components = self._get_nr_primary_components(primary_components, threshold=events.threshold)
+        scaling_factor = self._calculate_scaling_factor(len(events.keys), data_matrix.shape[1], nr_components)
+
+        return scaling_factor
+
+    @staticmethod
+    def _pca_analysis(data_matrix):
+        """
+        Performs PCA on matrix
+        """
+        pca = PCA()
+        pca.fit(data_matrix)
+        return pca
+
+    @staticmethod
+    def _get_nr_primary_components(pca, threshold):
```

Comment:
Avoid using sklearn objects as parameters, i.e., combine into one function possibly as we discussed